extensions/redis-client/deployment/src/main/java/io/quarkus/redis/client/deployment/DevServicesProcessor.java

```diff
@@ -0,0 +1,210 @@
+
+public class DevServicesProcessor {
+    private static final Logger log = Logger.getLogger(DevServicesProcessor.class);
+    private static final String REDIS_6_ALPINE = "redis:6-alpine";
+    private static final int REDIS_EXPOSED_PORT = 6379;
+    private static final String REDIS_SCHEME = "redis://";
+    private static final String QUARKUS = "quarkus.";
+    private static final String DOT = ".";
+    private static volatile List<Closeable> closeables;
+    private static volatile Map<String, DevServiceConfiguration> capturedDevServicesConfiguration;
+    private static volatile boolean first = true;
+
+    @BuildStep(onlyIfNot = IsNormal.class, onlyIf = DockerRunning.class)
+    public DevServicesRedisResultBuildItem startRedisContainers(LaunchModeBuildItem launchMode,
+            BuildProducer<RunTimeConfigurationDefaultBuildItem> runTimeConfiguration,
+            BuildProducer<ServiceStartBuildItem> serviceStartBuildItemBuildProducer, RedisBuildTimeConfig config) {
+
+        Map<String, DevServiceConfiguration> currentDevServicesConfiguration = new HashMap<>(config.additionalDevServices);
+        currentDevServicesConfiguration.put(RedisClientUtil.DEFAULT_CLIENT, config.defaultDevService);
+
+        // figure out if we need to shut down and restart existing redis containers
+        // if not and the redis containers have already started we just return
+        if (closeables != null) {
+            boolean restartRequired = launchMode.getLaunchMode() == LaunchMode.TEST;
+            if (!restartRequired) {
+                restartRequired = !currentDevServicesConfiguration.equals(capturedDevServicesConfiguration);
+            }
+            if (!restartRequired) {
+                return null;
+            }
+            for (Closeable closeable : closeables) {
+                try {
+                    closeable.close();
+                } catch (Throwable e) {
+                    log.error("Failed to stop redis container", e);
+                }
+            }
+            closeables = null;
+            capturedDevServicesConfiguration = null;
+        }
+
+        capturedDevServicesConfiguration = currentDevServicesConfiguration;
+        Result defaultConnectionResult = null;
+        List<Closeable> currentCloseables = new ArrayList<>();
+        Map<String, Result> namedConnectionResults = new HashMap<>();
+        for (Entry<String, DevServiceConfiguration> entry : currentDevServicesConfiguration.entrySet()) {
+            String connectionName = entry.getKey();
+            StartResult startResult = startContainer(connectionName, entry.getValue().devservices);
+            if (startResult == null) {
+                continue;
+            }
+            currentCloseables.add(startResult.closeable);
+            String configKey = getConfigPrefix(connectionName) + RedisConfig.HOSTS_CONFIG_NAME;
+            runTimeConfiguration.produce(new RunTimeConfigurationDefaultBuildItem(configKey, startResult.url));
```

Comment:
This does not work on restart due to https://github.com/quarkusio/quarkus/issues/17069